astropy/coordinates/tests/test_representation_arithmetic.py

```diff
@@ -0,0 +1,322 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+# TEST_UNICODE_LITERALS
+import functools
+import numpy as np
+
+from ... import units as u
+from .. import (PhysicsSphericalRepresentation, CartesianRepresentation,
+                CylindricalRepresentation, SphericalRepresentation,
+                UnitSphericalRepresentation, Longitude, Latitude)
+from ..angle_utilities import angular_separation
+from ...tests.helper import pytest, assert_quantity_allclose
+
+
+def assert_representation_allclose(actual, desired, rtol=1.e-7, atol=None,
+                                   **kwargs):
+    assert_quantity_allclose(actual.to_cartesian().xyz,
+                             desired.to_cartesian().xyz,
+                             rtol, atol, **kwargs)
+
+
+def representation_equal(first, second):
+    return functools.reduce(np.logical_and,
+                            (getattr(first, component) ==
+                             getattr(second, component)
+                             for component in first.components))
+
+class TestArithmetic():
+    """Test absolute value, multiplication, and division."""
+
+    def setup(self):
+        self.lon = Longitude(np.arange(0, 12.1, 2), u.hourangle)
```

Comment:
The numbers are not really arbitrary, they ensure that various combinations give integer distances... I'll add a note to that effect.